instaclustr/api_client.go

```diff
@@ -168,6 +168,21 @@ func (c *APIClient) CreateFirewallRule(data []byte, clusterID string) error {
 	return nil
 }
 
+func (c *APIClient) UpdateFirewallRule(data []byte, clusterID string) error {
+	url := fmt.Sprintf("%s/provisioning/v1/%s/firewallRules/", c.apiServerHostname, clusterID)
+	resp, err := c.MakeRequest(url, "PUT", data)
+	if err != nil {
+		return err
+	}
+	bodyText, err := ioutil.ReadAll(resp.Body)
```

Comment:
slight optimisation - could we move this inside the below if statement so we dont do this on success if its not needed

Reply:
Not as it is, 'cos the body Text is put in the  error message

Reply:
I think Mitch means to put the `ioutil.ReadAll(...)` in the if check below. Also, I think we should also handle checking the `err` var incase the body is unreadible:
```go
if resp.StatusCode != 202 {
	bodyText, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		bodyText = "<unparsable body>"
	}
	return errors.New(fmt.Sprintf("Status code: %d, message: %s", resp.StatusCode, bodyText))
}
```